src/SmartComponents/modules/reducers.js

```diff
@@ -6,9 +6,66 @@ const initialState = {
     selectedSystemIds: [],
     filterDropdownOpened: false,
     stateFilter: 'all',
-    factFilter: ''
+    factFilter: '',
+    totalFacts: undefined,
+    page: 1,
+    perPage: 10,
+    filteredData: {}
 };
 
+function getTotal(data, stateFilter, factFilter) {
+    let total = 0;
+
+    for (let i = 0; i < data.facts.length; i++) {
```

Comment:
Note this replicates the logic in `filterCompareData`. Can you call that function directly and take the length?